internal/datacoord/job_manager.go

```diff
@@ -0,0 +1,299 @@
+	allocator allocator.Allocator
+}
+
+func newJobManager(ctx context.Context,
+	mt *meta,
+	scheduler *taskScheduler,
+	allocator allocator.Allocator) *statsJobManager {
+	ctx, cancel := context.WithCancel(ctx)
+	return &statsJobManager{
+		ctx:       ctx,
+		cancel:    cancel,
+		loopWg:    sync.WaitGroup{},
+		mt:        mt,
+		scheduler: scheduler,
+		allocator: allocator,
+	}
+}
+
+func (jm *statsJobManager) Start() {
+	if Params.DataCoordCfg.EnableStatsTask.GetAsBool() {
+		jm.loopWg.Add(2)
+		go jm.triggerStatsTaskLoop()
+		go jm.cleanupStatsTasksLoop()
+	}
+}
+
+func (jm *statsJobManager) Stop() {
+	jm.cancel()
+	jm.loopWg.Wait()
+}
+
+func (jm *statsJobManager) triggerStatsTaskLoop() {
+	log.Info("start checkStatsTaskLoop...")
+	defer jm.loopWg.Done()
+
+	ticker := time.NewTicker(Params.DataCoordCfg.TaskCheckInterval.GetAsDuration(time.Second))
+	defer ticker.Stop()
+	for {
+		select {
+		case <-jm.ctx.Done():
+			log.Warn("DataCoord context done, exit checkStatsTaskLoop...")
+			return
+		case <-ticker.C:
+			jm.triggerSortStatsTask()
+			jm.triggerTextIndexStatsTask()
+			jm.triggerBM25StatsTask()
+
+		case segID := <-getStatsTaskChSingleton():
+			log.Info("receive new segment to trigger stats task", zap.Int64("segmentID", segID))
+			segment := jm.mt.GetSegment(segID)
+			if segment == nil {
+				log.Warn("segment is not exist, no need to do stats task", zap.Int64("segmentID", segID))
+				continue
+			}
+			// TODO @xiaocai2333 @bigsheeper: remove code after allow create stats task for importing segment
+			if segment.GetIsImporting() {
+				log.Info("segment is importing, skip stats task", zap.Int64("segmentID", segID))
+				select {
+				case getBuildIndexChSingleton() <- segID:
+				default:
+				}
+				continue
+			}
+			jm.createSortStatsTaskForSegment(segment)
+		}
+	}
+}
+
+func (jm *statsJobManager) triggerSortStatsTask() {
+	segments := jm.mt.SelectSegments(SegmentFilterFunc(func(seg *SegmentInfo) bool {
+		return isFlush(seg) && seg.GetLevel() != datapb.SegmentLevel_L0 && !seg.GetIsSorted()
+	}))
+	for _, segment := range segments {
+		if !segment.GetIsSorted() {
+			// TODO @xiaocai2333, @bigsheeper:
+			if segment.GetIsImporting() {
+				log.Warn("segment is importing, skip stats task, wait @bigsheeper support it")
+				continue
+			}
+			jm.createSortStatsTaskForSegment(segment)
+		}
+	}
+}
+
+func (jm *statsJobManager) createSortStatsTaskForSegment(segment *SegmentInfo) {
+	targetSegmentID, err := jm.allocator.AllocID(jm.ctx)
+	if err != nil {
+		log.Warn("allocID for segment stats task failed",
+			zap.Int64("segmentID", segment.GetID()), zap.Error(err))
+		return
+	}
+	if err := jm.SubmitStatsTask(segment.GetID(), targetSegmentID, indexpb.StatsSubJob_Sort, true); err != nil {
+		log.Warn("create stats task with sort for segment failed, wait for retry",
+			zap.Int64("segmentID", segment.GetID()), zap.Error(err))
+		return
+	}
+}
+
+func (jm *statsJobManager) enableMatch(field *schemapb.FieldSchema) bool {
```

Comment:
In pkg/util/typeutil/field_schema.go 
func (h *FieldSchemaHelper) EnableMatch() bool { }